bdai_ros2_wrappers/examples/background_action_example.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2023 Boston Dynamics AI Institute Inc.  All rights reserved.
+
+"""
+An example of a ROS 2 aware executable using process-wide machinery.
+
+Run with:
+
+```sh
+python3 examples/background_action_example.py
+```
+
+And follow the instructions on screen.
+"""
+
+import sys
+from typing import Any
+
+from example_interfaces.action import Fibonacci
+from rclpy.action.client import ActionClient
+from rclpy.action.server import ActionServer, ServerGoalHandle
+
+import bdai_ros2_wrappers.process as ros_process
+from bdai_ros2_wrappers.node import Node
+
+
+class MinimalActionServer(Node):
+    """A minimal ROS 2 node serving an example_interfaces.action.Fibonacci action."""
+
+    def __init__(self, node_name: str = "minimal_action_server", **kwargs: Any) -> None:
+        super().__init__(node_name, **kwargs)
+        self._action_server = ActionServer(self, Fibonacci, "compute_fibonacci_sequence", self.execute_callback)
+
+    def execute_callback(self, goal_handle: ServerGoalHandle) -> Fibonacci.Result:
+        sequence = [0, 1]
+
+        for i in range(1, goal_handle.request.order):
+            sequence.append(sequence[i] + sequence[i - 1])
+            feedback = Fibonacci.Feedback()
+            feedback.sequence = sequence
+            goal_handle.publish_feedback(feedback)
+        goal_handle.succeed()
+
+        result = Fibonacci.Result()
+        result.sequence = sequence
+        return result
+
+
+@ros_process.main()
+def main() -> None:
+    """
+    Example entrypoint.
+
+    It is configured almost as a prebaked ROS 2 aware process. That is, including a
+    process-wide (ie. globally accessible) node, an autoscaling multi-threaded executor
+    running in a background thread, automatic ``logging`` logs forwarding to the ROS 2
+    logging system, and implicit namespacing for all ROS 2 nodes loaded in it based on
+    the executable basename. A convenience for simple executables that need to use ROS 2
+    without having to worry about ROS 2 (or its fallout, e.g. accidental deadblocking).
+
+    When executed, an action server node is loaded, and the process-wide node is used to
+    setup an action client. This action client is then used to implement a simple console
+    application.
+    """
+    main.load(MinimalActionServer)
+
+    action_client = ActionClient(main.node, Fibonacci, "compute_fibonacci_sequence")
+    assert action_client.wait_for_server(timeout_sec=5)
+
+    while True:
```

Comment:
Should this be `True` or something that checks whether ros is still alive?  Sorry can't remember if ROS2 has that atm...

Reply:
We can also loop on context status, done in https://github.com/bdaiinstitute/ros_utilities/pull/32/commits/efb0db0d1ef57bdecc49aaff0dcbe9222defb1e3.